src/wallet3/default_daemon_comms.cpp

```diff
@@ -167,11 +169,17 @@ namespace wallet
           crypto::hash new_hash;
 
           if (not dc.skip_until("hash"))
+          {
+            oxen::log::warning(logcat, "bad response from rpc.get_height, key 'hash' missing");
             throw std::runtime_error("bad response from rpc.get_height, key 'hash' missing");
+          }
           new_hash = tools::make_from_guts<crypto::hash>(dc.consume_string_view());
 
           if (not dc.skip_until("height"))
+          {
+            oxen::log::warning(logcat, "bad response from rpc.get_height, key 'height' missing");
             throw std::runtime_error("bad response from rpc.get_height, key 'height' missing");
+          }
```

Comment:
There's a bunch of identical chunks of code like this in this file which are begging to be abstracted into a local static function.